src/main/java/com/spotify/reaper/service/SegmentGenerator.java

```diff
@@ -156,7 +157,7 @@ public static boolean greaterThanOrEqual(BigInteger a, BigInteger b) {
       BigInteger size = segment.span(RANGE_SIZE);
       total = total.add(size);
     }
-    if (!total.equals(RANGE_SIZE)) {
+    if (!total.equals(RANGE_SIZE) && !incrementalRepair) {
```

Comment:
The thing is that we are tricking reaper with incremental repair, as our repair unit is no longer segments but nodes, and generate one bogus segment per node. 
This check started to fail after I added a test scenario including incremental repair and as it was not relevant I chose to disable the check with incremental repair.